src/RTTComms.cpp

```diff
@@ -311,6 +333,7 @@ namespace BrainCloud
 #if (TARGET_OS_WATCH != 1)
         std::thread connectionThread([this]
         {
+            _rttConnectionStatus = RTTConnectionStatus::Connecting;
```

Comment:
put this before the thread. the time the thread starts, etc. maybe a few frames pass already.